### PR TITLE
Enable init-tools option to suppress non-error output from the console output

### PR DESCRIPTION
**Pull request by chcosta**
Add an argument option to init-tools so we can suppress displaying non-error output to the console.  Allows us to run quietly unless an error is encountered.

/cc @weshaggard @maririos

### CODE REVIEW

File: run.cmd [hunk 1]
```diff
@@ -17,12 +17,11 @@ if not defined VisualStudioVersion (
 set Platform=
 
 :: Restore the Tools directory
-call %~dp0init-tools.cmd
+call %~dp0init-tools.cmd -quiet
```
Comment (weshaggard) [inline run.cmd]: I would prefer we don't add such an option to the cmd (or the sh which looks like you overlooked). What happens if you instead just redirect to a log file?

Comment (chcosta) [inline run.cmd]: Sounds reasonable.  And yes,  I've focused on windows for a couple projects at the moment  and overlooked Unix.  Thanks for mentioning this

Comment (chcosta) [inline run.cmd]: It's slightly disconcerting to not get the reassuring "Tools are already initialized" or "Done initializing tools" messages, but that hand holding isn't truly necessary, and additional information is still available with a bit more work ("set _echo=on" or checking init-tools.log).  It's also difficult not to add the inverse switch, "verbose", but I understand and agree with your principal of reducing these scripts, not bloating them.

Comment (maririos) [inline run.cmd]: >I've focused on windows for a couple projects at the moment and overlooked Unix.

Does that means that you are going to update the Unix files in this PR too?

Comment (chcosta) [inline run.cmd]: Yes.

File: init-tools.cmd [hunk 2]
```diff
@@ -34,10 +34,10 @@ echo Running %0 > "%INIT_TOOLS_LOG%"
 
 if exist "%DOTNET_CMD%" goto :afterdotnetrestore
 
-echo Installing dotnet cli...
+echo Installing dotnet cli... >> "%INIT_TOOLS_LOG%"
 if NOT exist "%DOTNET_PATH%" mkdir "%DOTNET_PATH%"
 set /p DOTNET_VERSION=< "%~dp0DotnetCLIVersion.txt"
-set DOTNET_ZIP_NAME=dotnet-dev-win-x64.%DOTNET_VERSION%.zip
+set DOTNET_ZIP_NAME=dotnet-dev-win-x86.%DOTNET_VERSION%.zip
```
Comment (maririos) [inline init-tools.cmd]: Why are we changing the dotnet that we were using?

Comment (chcosta) [inline init-tools.cmd]: Great catch, that was solely meant for testing.

**Comment by chcosta** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test 
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Comment by maririos** [comment]
LGTM

**Comment by ericstj** [comment]
/cc @Petermarcu @weshaggard 
This change removed all console spew from init-tools.   Why are we doing that?

init-tools takes a significant amount of time to run, now devs see no feedback on the commandline.

**Comment by chcosta** [comment]
My initial change for this made the spew output toggle-able via a command-line switch but Wes has a strong desire to reduce bloat in these scripts.  

> It's slightly disconcerting to not get the reassuring "Tools are already initialized" or "Done initializing tools" messages, but that hand holding isn't truly necessary, and additional information is still available with a bit more work ("set _echo=on" or checking init-tools.log). It's also difficult not to add the inverse switch, "verbose", but I understand and agree with your principal of reducing these scripts, not bloating them.

The impetus for this change is enabling us to get pure output from the run tool - https://github.com/dotnet/buildtools/pull/980

**Comment by ericstj** [comment]
> The impetus for this change is enabling us to get pure output from the run tool

Seems to me like this linked PR is for option to run in quiet mode, why then would we make the default of init-tools to be quiet rather than have it be opt-in quiet?

**Comment by weshaggard** [comment]
@ericstj how much status to you think should be echoed out of the init-tools script? I want to try an avoid adding more logic into these cmd/sh scripts. 

@chcosta originally I was suggesting that you pipe the call to init-tools.cmd to a file if people wanted to make it quite.

Perhaps there is a good middle ground which is to make it quite for the normal case where it is already initialized but when it is doing work at least tell that it is doing work. so it doesn't just sit there appearing to hang.

**Comment by chcosta** [comment]
I see.  I misinterpreted our conversation believing that you preferred the output in init-tools itself simply be piped to a log.  

The middle ground, I expect, will still cause problems when we are trying to tie the produces / consumes ouput and handle inter-repo builds.

Reverting to the previous behavior and/or middle ground and piping the output when init-tools gets called is fine, and would mean you can get the output you want when running init-tools directly, but it would be silent when called via the run tool.  I would hate for people then to blame the run tool for being slow, but that seems inevitable unless we can speed up init-tools somehow.

**Comment by weshaggard** [comment]
>The middle ground, I expect, will still cause problems when we are trying to tie the produces / consumes ouput and handle inter-repo builds.

Are we expecting that we actually capture the results by piping the output to a file? I hope not. If we need to capture the output we should make that a parameter we pass to the command to write that to a file.

**Comment by chcosta** [comment]
The repoutil does exactly that.  This was meant to bring us to closer parity with the spec and how Roslyn is handling the produces / consumes api.  We're not yet to a point where anybody is actually using these api's in a meaningful way so we haven't decided how this will work.  We need to ensure that RepoUtil is following the same pattern, or vice-versa.

**Comment by Petermarcu** [comment]
The customer experience we have to have here is that when I run build, by default I should be told whats happening at least to the level that ensures me something is happening. Long pauses especially at the very beginning lead to confusion.

**Pull request merged**